vault/rollback.go

```diff
@@ -29,7 +29,12 @@ const (
 // is in-flight at any given time within a single seal/unseal phase.
 type RollbackManager struct {
 	logger *log.Logger
-	mounts *MountTable
+
+	// This gives the current mount table, plus a RWMutex that is
+	// locked for reading. It is up to the caller to RUnlock it
+	// when done with the mount table
+	mounts func() (*MountTable, *sync.RWMutex)
```

Comment:
Can we have this function instead just return a slice of mount paths? This way the locking can be done internally and the slice is left to the caller